### ISSUE TITLE
[JENKINS-22706] Warn that space in TEMP or TMP causes msysgit ssh credentials failure

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#2465]
<p>Trying to switch from HTTPS to SSH. Getting the following exception after switching (changed the link for this post):</p>

<p>Fetching upstream changes from git@​server:username/repository.git<br/>
using GIT_SSH to set credentials server ssh<br/>
FATAL: Failed to fetch from git@​server:username/repository.git<br/>
hudson.plugins.git.GitException: Failed to fetch from git@​server:username/repository.git<br/>
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:623)<br/>
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:855)<br/>
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:880)<br/>
	at org.jenkinsci.plugins.multiplescms.MultiSCM.checkout(MultiSCM.java:118)<br/>
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1251)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:605)<br/>
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:88)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:514)<br/>
	at hudson.model.Run.execute(Run.java:1703)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:231)<br/>
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress git@​server:username/repository.git +refs/heads/master:refs/remotes/origin/master" returned status code 128:<br/>
stdout: <br/>
stderr: 'C:\Program' is not recognized as an internal or external command,<br/>
operable program or batch file.<br/>
fatal: Could not read from remote repository.</p>

<p>Please make sure you have the correct access rights<br/>
and the repository exists.</p>

<p>	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1276)<br/>
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1146)<br/>
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$200(CliGitAPIImpl.java:87)<br/>
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:254)<br/>
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:621)<br/>
	... 11 more</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=flaviodonze">flaviodonze</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706" target="_blank">Warn that space in TEMP or TMP causes msysgit ssh credentials failure</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: git-client-plugin, git-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2014-09-06T14:57:20+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 6
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Trying to switch from HTTPS to SSH. Getting the following exception after switching (changed the link for this post):</p>

<p>Fetching upstream changes from git@server:username/repository.git
using GIT_SSH to set credentials server ssh
FATAL: Failed to fetch from git@server:username/repository.git
hudson.plugins.git.GitException: Failed to fetch from git@server:username/repository.git
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:623)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:855)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:880)
	at org.jenkinsci.plugins.multiplescms.MultiSCM.checkout(MultiSCM.java:118)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1251)
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:605)
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:88)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:514)
	at hudson.model.Run.execute(Run.java:1703)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress git@server:username/repository.git +refs/heads/master:refs/remotes/origin/master" returned status code 128:
stdout: 
stderr: 'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
fatal: Could not read from remote repository.</p>

<p>Please make sure you have the correct access rights
and the repository exists.</p>

<p>	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1276)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1146)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$200(CliGitAPIImpl.java:87)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:254)
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:621)
	... 11 more</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>git 2.2.1, git-client 1.8.0, windows server 2008 R2 64bit</code></li></ul>
<details><summary><i>3 attachments</i></summary>

- [Configuration.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/25714/Configuration.PNG)
  > ![Configuration.PNG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/25714/Configuration.PNG)
- [errorJenkins.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/25711/errorJenkins.jpg)
  > ![errorJenkins.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/25711/errorJenkins.jpg)
- [hudson.plugins.git.GitTool.xml](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/25712/hudson.plugins.git.GitTool.xml)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199078&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199078">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I believe you will have better results if you define a credential with the user name and password (or with the ssh key if available), rather than embedding the user name and password in the ssh URL.</p>

<p>I'm perplexed why the warning appears about "C:\Program" not being a recognized command.  If that message continues even after you define a credential, you may want to check that Git\bin is in the PATH on the slave, or that the full path to the git executable is listed in the node configuration.</p>
</pre>
</details>

</ul>
</details>

<p>I believe you will have better results if you define a credential with the user name and password (or with the ssh key if available), rather than embedding the user name and password in the ssh URL.</p>

<p>I'm perplexed why the warning appears about "C:\Program" not being a recognized command.  If that message continues even after you define a credential, you may want to check that Git\bin is in the PATH on the slave, or that the full path to the git executable is listed in the node configuration.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199079&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199079">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Mark: Possibly related to <a href="http://www.oracle.com/technetwork/java/javase/7u21-relnotes-1932873.html#jruntime" class="external-link" target="_blank" rel="nofollow noopener">this</a>?</p>
</pre>
</details>

</ul>
</details>

<p>Mark: Possibly related to <a href="http://www.oracle.com/technetwork/java/javase/7u21-relnotes-1932873.html#jruntime" class="external-link" target="_blank" rel="nofollow noopener">this</a>?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>flaviodonze:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199100&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199100">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Mark, 
The password is not part of the URL, I made a mistake the URL does not contain the username but the user who created the repository on bitbucket.
So there is no login data in the URL. git@bitbucket.org:hosteduser/repository.git</p>

<p>What confuses me, git fetching works using HTTPS (having a defined jenkins credentials) and when using SSH "'C:\Program' is not recognized" is thrown.</p>

<p>I switched the environment variables to GIT_HOME=C:\Progra~2\Git\bin or GIT_SSH=C:\Progra~1\TortoiseGit\bin\TortoisePlink.exe
But it seems that some other variable is used and I can not find it. </p>
</pre>
</details>

</ul>
</details>

<p>Hi Mark, <br/>
The password is not part of the URL, I made a mistake the URL does not contain the username but the user who created the repository on bitbucket.<br/>
So there is no login data in the URL. git@​bitbucket.org:hosteduser/repository.git</p>

<p>What confuses me, git fetching works using HTTPS (having a defined jenkins credentials) and when using SSH "'C:\Program' is not recognized" is thrown.</p>

<p>I switched the environment variables to GIT_HOME=C:\Progra~2\Git\bin or GIT_SSH=C:\Progra~1\TortoiseGit\bin\TortoisePlink.exe<br/>
But it seems that some other variable is used and I can not find it. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>deepak:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199106&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199106">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,</p>

<blockquote><p>stdout:
stderr: 'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
fatal: Could not read from remote repository.</p></blockquote>

<p>looks pretty much like commandline error. If i'm not mistaken the plugin looks to have not been able to reach the git binary. Is the git installed in the following directory?</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">c:\Program Files</pre>
</div></div>
<p>As a temporary workaround i suggest installing git in a directory without spaces and see if it works(just the first thought that bubbled up in my mind)</p>

<p>Putting the git installation path in a set of double-quotes also might help.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">"</pre>
</div></div>

<p><b><font color="red">I am having problem on a unix installation of jenkins.</font></b></p>

<p>The error looks something like this</p>

<p><b><blockquote><p><font color="maroon">
Failed to connect to repository : Command "git ls-remote -h devcm:/cmr/Platform/Platform.git HEAD" returned status code 128:
stdout:
stderr: Permission denied, please try again.
Permission denied, please try again.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.</font></p></blockquote></b></p>


<p><b><font color="blue">This behavior is seen in shell when you type in the git clone command and press enter without keying in the password.</font></b>
i.e you just press enter at the password prompt without entering anything</p>

<p>I'm using RHEL6 environment</p>

<p>Any help will be much appreciated</p>

<p>Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi,</p>

<blockquote><p>stdout:<br/>
stderr: 'C:\Program' is not recognized as an internal or external command,<br/>
operable program or batch file.<br/>
fatal: Could not read from remote repository.</p></blockquote>

<p>looks pretty much like commandline error. If i'm not mistaken the plugin looks to have not been able to reach the git binary. Is the git installed in the following directory?</p>

<pre>
c:\Program Files</pre>
<p>As a temporary workaround i suggest installing git in a directory without spaces and see if it works(just the first thought that bubbled up in my mind)</p>

<p>Putting the git installation path in a set of double-quotes also might help.</p>

<pre>
"</pre>

<p><b><font color="red">I am having problem on a unix installation of jenkins.</font></b></p>

<p>The error looks something like this</p>

<p><b><blockquote><p><font color="maroon"><br/>
Failed to connect to repository : Command "git ls-remote -h devcm:/cmr/Platform/Platform.git HEAD" returned status code 128:<br/>
stdout:<br/>
stderr: Permission denied, please try again.<br/>
Permission denied, please try again.<br/>
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).<br/>
fatal: Could not read from remote repository.<br/>
<br/>
Please make sure you have the correct access rights<br/>
and the repository exists.</font></p></blockquote></b></p>


<p><b><font color="blue">This behavior is seen in shell when you type in the git clone command and press enter without keying in the password.</font></b><br/>
i.e you just press enter at the password prompt without entering anything</p>

<p>I'm using RHEL6 environment</p>

<p>Any help will be much appreciated</p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>deepak:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199107&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199107">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Mark,</p>

<p>I've tried all combinations of credentials. I've also gone to lengths of creating more users and using those credentials.</p>

<p>As a final resort, i even tried setting password for the jenkins user as it was the only user with access to workspace. Then i tried a different user and modified the permissions of workspace.</p>

<p>Nothing seems to work...</p>
</pre>
</details>

</ul>
</details>

<p>Hi Mark,</p>

<p>I've tried all combinations of credentials. I've also gone to lengths of creating more users and using those credentials.</p>

<p>As a final resort, i even tried setting password for the jenkins user as it was the only user with access to workspace. Then i tried a different user and modified the permissions of workspace.</p>

<p>Nothing seems to work...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>deepak:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199109&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199109">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Attached a screenshot of the error</p>
</pre>
</details>

</ul>
</details>

<p>Attached a screenshot of the error</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199118&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199118">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Deepak, your problem may be related to the version of git which is installed by default on RHEL6.  I believe the default git version on RHEL is 1.7.1.  A minimum version of 1.7.10 is required for credentials support.  You might check your git version.</p>
</pre>
</details>

</ul>
</details>

<p>Deepak, your problem may be related to the version of git which is installed by default on RHEL6.  I believe the default git version on RHEL is 1.7.1.  A minimum version of 1.7.10 is required for credentials support.  You might check your git version.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199119&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199119">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=flaviodonze" class="user-hover" rel="flaviodonze">flaviodonze</a> Could you upload your system-wide Jenkins configuration file (usually stored in the Jenkins home directory as config.xml)?  I'd like to see the exact definition of the location of the git program.  Mine is "D:\Program Files (x86)\git\bin\git.exe" for one of my slaves and is not set of the other slaves (relying on git.exe being found in the system path).</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=flaviodonze" class="user-hover" rel="flaviodonze">flaviodonze</a> Could you upload your system-wide Jenkins configuration file (usually stored in the Jenkins home directory as config.xml)?  I'd like to see the exact definition of the location of the git program.  Mine is "D:\Program Files (x86)\git\bin\git.exe" for one of my slaves and is not set of the other slaves (relying on git.exe being found in the system path).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>flaviodonze:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199124&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199124">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Couldn't find anything git related in the config.xml, found the hudson.plugins.git.GitTool.xml though. Is this what you are asking for?</p>

<p>Trying to fix the issue on my own I already had changed the configuration to "C:/Progra~2/Git/bin/git.exe" in the jenkins admin area.</p>
</pre>
</details>

</ul>
</details>

<p>Couldn't find anything git related in the config.xml, found the hudson.plugins.git.GitTool.xml though. Is this what you are asking for?</p>

<p>Trying to fix the issue on my own I already had changed the configuration to "C:/Progra~2/Git/bin/git.exe" in the jenkins admin area.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199125&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199125">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The git location can be specified in the Jenkins configuration for the slave as a tool.  If that location were incorrect, it might cause the behavior you're seeing.</p>
</pre>
</details>

</ul>
</details>

<p>The git location can be specified in the Jenkins configuration for the slave as a tool.  If that location were incorrect, it might cause the behavior you're seeing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>flaviodonze:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199127">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is my configuration.</p>
</pre>
</details>

</ul>
</details>

<p>This is my configuration.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>deepak:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=199345&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199345">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Mark,
Thanks for the suggestion.</p>

<p>I had already upgraded git to version 1.9.0 before starting with my repository setup.</p>
<blockquote><p>Deepak, your problem may be related to the version of git which is installed by default on RHEL6. I believe the default git version on RHEL is 1.7.1. A minimum version of 1.7.10 is required for credentials support. You might check your git version.</p></blockquote>

<p>No harm in checking again,</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"># git --version
git version 1.9.0</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Mark,<br/>
Thanks for the suggestion.</p>

<p>I had already upgraded git to version 1.9.0 before starting with my repository setup.</p>
<blockquote><p>Deepak, your problem may be related to the version of git which is installed by default on RHEL6. I believe the default git version on RHEL is 1.7.1. A minimum version of 1.7.10 is required for credentials support. You might check your git version.</p></blockquote>

<p>No harm in checking again,</p>

<pre>
# git --version
git version 1.9.0</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/estyrke.jpg" title="estyrke's avatar" /> estyrke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=207953&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-207953">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>As per <a href="https://issues.jenkins-ci.org/browse/JENKINS-21030?focusedCommentId=193862&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-193862" class="external-link" target="_blank" rel="nofollow noopener">this comment</a>, the problem may be that java.io.tmpdir contains spaces. Setting this variable to "C:\temp" fixed this problem for me.</p>
</pre>
</details>

</ul>
</details>

<p>As per <a href="https://issues.jenkins-ci.org/browse/JENKINS-21030?focusedCommentId=193862&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-193862" class="external-link" target="_blank" rel="nofollow noopener">this comment</a>, the problem may be that java.io.tmpdir contains spaces. Setting this variable to "C:\temp" fixed this problem for me.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=207982&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-207982">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=estyrke" class="user-hover" rel="estyrke">estyrke</a> for detecting that unexpected dependency which msysgit has on a temporary directory path which does not include spaces.  I've confirmed the same behavior.  If I set TEMP=C:\has a space\tmp and have that in the environment when I run my windows slave, then this exact failure appears when cloning from an ssh git URL.</p>

<p>I'll consider what alternatives are available to at least inform the user from the git client plugin when such a problem happens.  The work around is straightforward, but the messages in the failure do not give any hint that is the work around.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=estyrke" class="user-hover" rel="estyrke">estyrke</a> for detecting that unexpected dependency which msysgit has on a temporary directory path which does not include spaces.  I've confirmed the same behavior.  If I set TEMP=C:\has a space\tmp and have that in the environment when I run my windows slave, then this exact failure appears when cloning from an ssh git URL.</p>

<p>I'll consider what alternatives are available to at least inform the user from the git client plugin when such a problem happens.  The work around is straightforward, but the messages in the failure do not give any hint that is the work around.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=209984&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-209984">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Mark Waite
Path:
 src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java
 src/test/java/org/jenkinsci/plugins/gitclient/WarnTempDirValueTest.java
<a href="http://jenkins-ci.org/commit/git-client-plugin/96f9c8bd9c1229bfe644e0a30356208890a53696" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/96f9c8bd9c1229bfe644e0a30356208890a53696</a>
Log:
  <span class="error">&#91;Fix JENKINS-22706&#93;</span> Warn Windows users if TEMP or TMP value contains embedded space</p>

<p>Some msysgit internal credentials operations fail with a message that
"C:\Program" is not a valid path if the TEMP or TMP directory name
contains an embedded space.</p>

<p>This warning is intended to help the user diagnose and repair their
configuration error.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Mark Waite<br/>
Path:<br/>
 src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java<br/>
 src/test/java/org/jenkinsci/plugins/gitclient/WarnTempDirValueTest.java<br/>
<a href="http://jenkins-ci.org/commit/git-client-plugin/96f9c8bd9c1229bfe644e0a30356208890a53696" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-client-plugin/96f9c8bd9c1229bfe644e0a30356208890a53696</a><br/>
Log:<br/>
  <span class="error">&#91;Fix JENKINS-22706&#93;</span> Warn Windows users if TEMP or TMP value contains embedded space</p>

<p>Some msysgit internal credentials operations fail with a message that<br/>
"C:\Program" is not a valid path if the TEMP or TMP directory name<br/>
contains an embedded space.</p>

<p>This warning is intended to help the user diagnose and repair their<br/>
configuration error.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=210366&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-210366">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Warning was added in git-client-plugin 1.10.2, released 12 Sep 2014</p>
</pre>
</details>

</ul>
</details>

<p>Warning was added in git-client-plugin 1.10.2, released 12 Sep 2014</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nitroinside:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=211295&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211295">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>In my case the error happened because Apache Temp Dir was setted to Program Files (x86). So, the solution for me was replace in Java args for Tomcat:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">-Djava.io.tmpdir=C:\Program Files (x86)\Apache Software Foundation\Tomcat 7.0\temp
</pre>
</div></div>

<p>To: </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">-Djava.io.tmpdir=C:\Windows\Temp
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>In my case the error happened because Apache Temp Dir was setted to Program Files (x86). So, the solution for me was replace in Java args for Tomcat:</p>


<pre>
-Djava.io.tmpdir=C:\Program Files (x86)\Apache Software Foundation\Tomcat 7.0\temp
</pre>

<p>To: </p>


<pre>
-Djava.io.tmpdir=C:\Windows\Temp
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=211296&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211296">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nitroinside" class="user-hover" rel="nitroinside">nitroinside</a> did the newly added warning actually appear in your case?  If the change was only to the value of java.io.tmpdir, I would expect it to not appear, since the warning message is displayed for environment variables with spaces in their value, not for Java properties with spaces in the value.</p>

<p>Though, I guess the Java virtual machine must be communicating the value of java.io.tmpdir to the git subprocess somehow, and that communication is probably through environment variables.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nitroinside" class="user-hover" rel="nitroinside">nitroinside</a> did the newly added warning actually appear in your case?  If the change was only to the value of java.io.tmpdir, I would expect it to not appear, since the warning message is displayed for environment variables with spaces in their value, not for Java properties with spaces in the value.</p>

<p>Though, I guess the Java virtual machine must be communicating the value of java.io.tmpdir to the git subprocess somehow, and that communication is probably through environment variables.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nitroinside:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=211298&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211298">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> no, the warning didn't appear, just failed when executing <em>fetch</em> command as the description in this issue. I think it must be a buggy parsing of the argument or something like that (I haven't seen the code), because in a previous version of the plugin with the same TEMP variable the error didn't occured.</p>

<p>Also, I tried to set a environment variable in Jenkins configuration:</p>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Name=TEMP
Value=C:\Windows\Temp
</pre>
</div></div>

<p>But was ignored and the error still. I assume that code takes the virtual machine's value instead the configuration's.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> no, the warning didn't appear, just failed when executing <em>fetch</em> command as the description in this issue. I think it must be a buggy parsing of the argument or something like that (I haven't seen the code), because in a previous version of the plugin with the same TEMP variable the error didn't occured.</p>

<p>Also, I tried to set a environment variable in Jenkins configuration:</p>



<pre>
Name=TEMP
Value=C:\Windows\Temp
</pre>

<p>But was ignored and the error still. I assume that code takes the virtual machine's value instead the configuration's.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=211299&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211299">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nitroinside" class="user-hover" rel="nitroinside">nitroinside</a> as far as I could tell from my reviews of the code, the problem was not in the plugin, but in the git command line implementation.  </p>

<p>There have been changes to the usage of the git command line calls, in particular because the git plugin now allows use of the Jenkins credentials plugin, while the original versions of the plugin did not support credential management from within Jenkins.</p>

<p>If you find a case in the source code where you think a mistake is being made, pull requests are certainly welcomed.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nitroinside" class="user-hover" rel="nitroinside">nitroinside</a> as far as I could tell from my reviews of the code, the problem was not in the plugin, but in the git command line implementation.  </p>

<p>There have been changes to the usage of the git command line calls, in particular because the git plugin now allows use of the Jenkins credentials plugin, while the original versions of the plugin did not support credential management from within Jenkins.</p>

<p>If you find a case in the source code where you think a mistake is being made, pull requests are certainly welcomed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nitroinside:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-22706?focusedId=211301&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211301">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> thanks for the quick answer! I guess you're right, probably is because the command line calls.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> thanks for the quick answer! I guess you're right, probably is because the command line calls.</p>
